.github/workflows/docs.yml

```diff
@@ -44,6 +44,29 @@ jobs:
         working-directory: ${{ steps.set-up-homebrew.outputs.repository-path }}/docs
         run: bundle exec rake build
 
+  links:
+    if: github.repository == 'Homebrew/brew'
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+        with:
+          core: false
+          cask: false
+          test-bot: false
+
+      - name: Install Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "3.1"
+          bundler-cache: true
+          working-directory: ${{ steps.set-up-homebrew.outputs.repository-path }}/docs
+
+      - name: Check for broken links in docs
```

Comment:
Maybe worth rolling this in under `Build docs site` in `linting` as it's going to be building in here anyway and doesn't seem too slow?

Reply:
HTML Proofer needs Ruby 3 and the rest of the `linting` job runs on Ruby 2.7. Do we mind if we bump that up?

Reply:
Go for it!